lib/Mojo/UserAgent/Role/Cache.pm

```diff
@@ -11,6 +11,21 @@ our $VERSION = '0.01';
 my $DEFAULT_STRATEGY = 'playback_or_record';
 
 has cache_driver => sub { shift->cache_driver_singleton };
+
+has cache_key => sub {
+  return sub {
+    my $req  = shift->req;
+    my $url  = $req->url;
+    my @keys = (lc $req->method);
+
+    push @keys, $url->host // '';
+    push @keys, $url->path_query;
+    push @keys, Mojo::Util::md5_sum($req->body) unless $keys[0] eq 'get';
```

Comment:
I wonder if it wouldn't be better to do if $req->body here instead of checking for 'get'? to avoid the md5 sum for a head or delete without body for instance?